
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Documentation</title>
    
    <meta name="author" content="Ilya Y. Zhbannikov">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="../css/bootstrap.min.css" rel="stylesheet">
    <link href="../css/style.css?body=1" rel="stylesheet" type="text/css" media="all">
    <link href="../css/kbroman.css" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
    
  </head>

  	<div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <a class="brand" href="http://izhbannikov.github.io/MetAmp">Meta-Amplicon Analysis Project</a>
          <a href="http://www.ibest.uiaho.edu"><img src="../media/ibest_logo.jpg" width=100 height=30 alt="Site logo" align = "right" style="opacity:0.4;"></a>
          <ul class="nav">
              <li><a href="http://izhbannikov.github.io/MetAmp/">Home</a></li>
              <li><a href="https://github.com/izhbannikov/MetAmp/zipball/master" id="download-zip" class="button">Download</a></li>
              <li><a href="../pages/documentation.html">Documentation</a></li>
              <li><a href="../pages/resources.html">Resources</a></li>
          </ul>
        </div>
      </div>
    </div>
    
    <div class="container-narrow">

      <div class="content">
        
<div class="page-header">
  <h2>Documentation</h2>
</div>

<div class="row-fluid">
  <div class="span12">
  	<p>Documentation in one file: <a href="../media/UserGuide.pdf">UserGuide.pdf</a> </p>
    <h3>Program description</h3>
    <p>MetAmp tool is developed for analysis of amplicon data by combining several marker regions from 16S rRNA genes. Such marker regions serve as unique identificators for species. There are nine marker regions in bacterial 16S rRNA gene.</p>
	
	<h3>Installation</h3>
	<p>Make sure that you have a stable Internet connection and R, Python 2.7.6 (at least, since older versions may not support some features) and GCC installed. Download or clone from the GitHub, save it in some folder you wish. Then simply run make. This will check for required packages and install in case if some packages were not found.</p>
	<pre>$cd MetAmp_home_directory
$make</pre>
	<p>In some cases you have to be a system administrator, so you need to compile MetAmp with the following command:</p>
	<pre>$sudo make</pre>
	
	<h3>Quick start</h3>
	
	<p>
	Before analysis, you may need to perform re-labeling of your read headers: a read header should contain barcodelabel (see more at www.drive5.com), for example:

<pre>@read1;<strong>barcodelabel=TCAG</strong>;
GATGAACGCTGGCGGCGTGCCTAATACATGCAAGT...AT
+
IIIAAAIIIIIIIIIIIIIIIHHHIAAAAAAAAAA...II</pre>

If you use Illumina sequence data, you also have to merge overlapping paired-end reads.

Later I will provide the scripts that can do these things above, but for now you can simply use scripts provided at 
<a href="http://drive5.com">www.drive5.com</a>. 
	</p>
	
	<p>To run the program, you have to provide reference libraries, your amplicon data and output directory where all analysis results will be strored.	</p>
	<p><strong>One marker region:</strong></p>
		<pre>$python metamp.py -r data/gold21/gold21.fasta -r1 data/gold21/gold21_V13V31.fasta -l1 data/even/SRR072220_V13V31_relabeled.fastq -o test</pre>
			Here:
			<ul>
				<p><code>metamp.py</code> - a program name.</p>
				<p><code>-r data/gold21/gold21.fasta</code> - passing the reference file, that contains whole 16S sequences, in forward and reverse complement.</p>
				<p><code>-r1 data/gold21/gold21_V13V31.fasta</code> - passing the reference file that contains marker (region V1-3) sequences extracted from gold21.fasta (whole 16S), in forward and reverse complement.</p>
				<p><code>-l1 data/even/SRR072220_V13V31_relabeled.fastq</code> - amplicon emprirical reads. ”relabeled” means that a barcode sequences were attached to each read label.</p>
				<p><code>-o test</code> - output directory with all analysis results.</p>
			</ul>
		<p><strong>Three markers:</strong></p>
			<pre>$python metamp.py -r data/gold21/gold21.fasta -r1 data/gold21/gold21_V13V31.fasta -l1 data/even/SRR072220_V13V31_relabeled.fastq -r2 data/gold21/gold21_V35V53.fasta -l2 data/even/SRR072220_V35V53_relabeled.fastq -r3 data/gold21/gold21_V69V96.fasta -l3 data/even/SRR072239_V69V96_relabeled.fastq -o test</pre>
			<p>Here:</p>
			<ul>
			<p><code>-r2 data/gold21/gold21_V35V53.fasta</code> - a reference file that contains marker (re- gion V3-5) sequences extracted from gold21.fasta (whole 16S), in forward and reverse com- plement.</p>
				<p><code>-l2 data/even/SRR072220_V35V53_relabeled.fastq</code> - amplicon empirical reads.</p>
				<p><codee>-r3 data/gold21/gold21_V69V96.fasta</code>- a reference file that contains marker (re- gion V3-5) sequences extracted from gold21.fasta (whole 16S), in forward and reverse com- plement.</p>
				<p><code>-l3 data/even/SRR072239_V69V96_relabeled.fastq</code> - amplicon empirical reads.</p>
			</ul>
		<p>To get help from the program, you can run <code>python metamp.py -h</code></p>
		<p>Output files will be in analysis directory that you provided with the -o command:</p>
			<ul>			
    			<p><samp>clusters.clstr</samp> - contains large table in uc format (see www.drive5.com).</p>
    			<p><samp>otu_table.txt</samp> - OTU table, where rows are OTUs and colums are barcodes.</p>
    			<p><samp>coordinates.crd</samp> - file that contains NMDS coordinates of each point (reference and empirical, see provided explanations in UserGuide.pdf).</p>
    			<p><samp>log.txt</samp> - a simple log file that record every stage of analysis.</p>
			</ul>
	</div>
</div>


      </div>
      <hr>
      <footer>
        <p><small>
  <!-- start of MetAmp's footer; modify this part -->
              <a href="http://www.uidaho.edu"> &copy; 2014 University of Idaho</a>
  <!-- end of MetAmp's footer; modify this part -->
        </small></p>
      </footer>

    </div>

    
  </body>
</html>

